drivers/adc-dac/ad74413r/ad74413r.c

```diff
@@ -156,7 +156,7 @@ int ad74413r_dac_voltage_to_code(uint32_t mvolts, uint32_t *code)
  * @param val - Register value.
  * @param buff - The communication buffer.
  */
-static void ad74413r_format_reg_write(uint8_t reg, uint16_t val, uint8_t *buff)
```

Comment:
the data width for this device is exactly 16-bit, according to datasheet, therefore we already reduce the risk considerably of sending random data that might be passed with 32-bit wide values when not properly initialized. Moreover, is it a good idea to adapt a driver based on the upper layers? It should be vice-versa from my point of view.

Reply:
for this exact cast there is no risk of sending random data, since the `no_os_put_unaligned_be16()` takes a `uint16_t` as a parameter, thus it will cast it to 16 bit (meaning that the upper 16 bits will just be ignored). However I agree that since this function doesn't receive a pointer to the `val` parameter there is no good reason for this change and I can revert it. The behavior is the same in both cases.

What I meant by "requiring extra handling in the upper layers" is the following case: `ad74413r_get_adc_range(struct ad74413r_desc *, uint32_t, uint16_t *)` -  this function has to receive a pointer to a 16 bit wide integer. Passing a pointer to `uint32_t` on a little endian cpu will result in writing at the wrong address. I think it would be a better API to just use pointers to 32 bit integers for all the cases where a value has to be read.

Reply:
I think this 16 -> 32 bits change may introduce bugs

For example line 162 here, i think it might not work on any endianess because it's now picking 2 bytes out of 4 and rearranges them. With val being 16 bits, it would work as intended, picking 2 bytes out of 2 (all good!) and rearranging them... 
https://github.com/analogdevicesinc/no-OS/blob/a78abcb369f233b9086c0ccf8ae0287e761f4648/drivers/adc-dac/ad74413r/ad74413r.c#L159-L164